frontend/src/app/components/resizer/wp-resizer.component.ts

```diff
@@ -136,10 +149,32 @@ export class WpResizerDirective extends UntilDestroyedMixin implements OnInit, A
   }
 
   resizeMove(deltas:ResizeDelta) {
+    // Avoid text-selection while the user is dragging the resizer
+    deltas.origin.preventDefault();
+
     // Get new value depending on the delta
-    // The resizingElement is not allowed to be smaller than 530px
     this.elementWidth = this.elementWidth - deltas.relative.x;
-    let newValue = this.elementWidth < 530 ? 530 : this.elementWidth;
+    let newValue;
+
+    // The resizingElement is not allowed to be smaller than the elementMinWidth
+    if (this.elementWidth < this.elementMinWidth) {
+      newValue = this.elementMinWidth;
+
+      // Show the resizer red when it reaches its limit (min-width)
+      if (this.resizer.style.color !== 'red') {
+        this.resizer.style.color = 'red';
+
+        // Avoid triggering change detection for all the app
+        this.ngZone.runOutsideAngular(() => {
```

Comment:
Without having looked at this further, can't we make this component `OnPush` and avoid these tricks?

Reply:
Hi @oliverguenther ,
Thank you for your feedback.

This common performance optimization utility (runOutsideAngular) is aimed to avoid triggering Change Detection for the entire app, in this case just for updating one component's view. Using OnPush is an improvement (just applied, thanks), but it wouldn't avoid this extra Change Detection caused by the setTimeout as this sample illustrates: https://stackblitz.com/edit/change-detection-onpush-timeout-triggers-app-change-detection?file=src/app/child/child.component.ts.

Maybe this kind of performance optimization is not necessary on this app right now, I made it just in case.

Looking forward to hearing your thoughts on this.

Reply:
Hi @aleixsuau, thanks for your explanation. First-off, welcome to the team and glad you're here! :partying_face: 

You're right that this avoids an extra cycle. With `onPush`, that at least gets reduced to this component + tree so it should be negligible.

I'm fine with leaving it in. That being said, we did not use this pattern much so far as it only works for direct DOM changes and in this case, I would have written in differently - e.g., with switching a variable to apply a conditional class. I think this is also relevant to Henriettes remark.

Does this make sense?

Reply:
>  First-off, I missed this in the previous comment. Welcome to the team and glad you're here! 🥳
Oh my good, I am sorry! I missed that, too 🙈 Welcome on board 🥳

Reply:
Thank you very much for the welcoming @oliverguenther and @HDinger, I'm really excited about becoming part this project :) 

I've made a few changes that could solve the issues you pointed out.

> with switching a variable to apply a conditional class
@oliverguenther Do you mean using ngClass or similar? Since the element we are styling is placed outside this component, the only way that came to my mind was using the DOM API.

Reply:
Thanks for the explanation, I wasn't aware that this styled outside of its component, then this of course is all perfectly reasonable. 👍

Reply:
Great, thanks 👍